darts/models/forecasting/prophet_model.py

```diff
@@ -220,6 +244,22 @@ def _predict(
         verbose: bool = False,
     ) -> TimeSeries:
 
+        for seasonality_name, attributes in self._add_seasonalities.items():
+            if attributes["condition_name"] is not None:
+                raise_if(
+                    future_covariates is None,
+                    f"Condition name '{attributes['condition_name']}' is required by "
+                    f"the custom seasonality '{seasonality_name}', but future_covariates is None. In addition, "
+                    f"the model should be re-trained with future_covariates.",
+                    logger,
+                )
```

Comment:
This will be raised in case a user, for whatever reason, fits a model, calls add_seasonality() after fitting and then predict(). Hence the "model should be retrained with future_covariates" message.

Reply:
The error is now raised according to the logic in `_check_seasonality_conditions()`